src/AlgebraicPetri.jl

```diff
@@ -1,186 +1,297 @@
 """ Computing in the category of finite sets and Petri cospans
 """
 module AlgebraicPetri
-export PetriCospanOb, PetriFunctor, PetriCospan
 
-using Catlab.GAT
-using Catlab.Theories: BiproductCategory
-using Catlab.CategoricalAlgebra.FreeDiagrams
-using Catlab.Theories
-using Catlab.CategoricalAlgebra.Limits
+export TheoryPetriNet, PetriNet, OpenPetriNetOb, AbstractPetriNet, ns, nt, ni, no,
+  add_species!, add_transition!, add_transitions!,
+  add_input!, add_inputs!, add_output!, add_outputs!, inputs, outputs,
+  TransitionMatrices, vectorfield,
+  TheoryLabelledPetriNet, LabelledPetriNet, AbstractLabelledPetriNet, sname, tname,
+  TheoryReactionNet, ReactionNet, AbstractReactionNet, concentration, concentrations, rate, rates,
+  TheoryLabelledReactionNet, LabelledReactionNet, AbstractLabelledReactionNet,
+  Open, OpenPetriNet, OpenLabelledPetriNet, OpenReactionNet, OpenLabelledReactionNet,
+  OpenPetriNetOb, OpenLabelledPetriNetOb, OpenReactionNetOb, OpenLabelledReactionNetOb
+
+using Catlab
+using Catlab.CategoricalAlgebra
 using Catlab.CategoricalAlgebra.FinSets
-using AutoHashEquals
+using Catlab.Present
+using Catlab.Theories
+using Petri
+using LabelledArrays
+using LinearAlgebra: mul!
+import Petri: Model, vectorfield
+
+vectorify(n) = begin
```

Comment:
should this be a constructor for Base.Vector and use multiple dispatch instead of the type switch logic below?

Reply:
Fixed in the latest commit